tools/testsys/src/run.rs

```diff
@@ -0,0 +1,204 @@
+        // config.
+        let region = if let Some(region) = self.target_region {
+            debug!("Using provided region for testing");
+            region
+        } else {
+            debug!("No region was provided, determining region from `Infra.toml`");
+            aws.regions
+                .clone()
+                .pop_front()
+                .context("No region was provided and no regions found in infra config")?
+        };
+
+        match variant.family() {
+            "aws-k8s" => {
+                debug!("Variant is in 'aws-k8s' family");
+                let file = File::open(&self.ami_input).context("Unable to open amis.json")?;
+                let ami_input: HashMap<String, Image> = serde_json::from_reader(file)
+                    .context(format!("Unable to deserialize '{}'", self.ami_input))?;
+                ensure!(!ami_input.is_empty(), "amis.json is empty");
+                let bottlerocket_ami = &ami_input
+                    .get(&region)
+                    .context(format!("ami not found for region '{}'", region))?
+                    .id;
+                debug!("Using ami '{}'", bottlerocket_ami);
+                let aws_k8s = AwsK8s {
+                    arch: self.arch,
+                    variant: self.variant,
+                    region,
+                    assume_role: self.assume_role,
+                    instance_type: self.instance_type,
+                    ami: bottlerocket_ami.to_string(),
+                    secrets,
+                    kube_conformance_image: self.kube_conformance_image,
+                    target_cluster_name: self.target_cluster_name,
+                };
+                debug!("Creating crds for aws-k8s testing");
+                let crds = aws_k8s.create_crds(self.test_flavor, &self.agent_images)?;
+                debug!("Adding crds to testsys cluster");
+                for crd in crds {
+                    let crd = client
+                        .create_object(crd)
+                        .await
+                        .context("Unable to create object")?;
+                    if let Some(name) = crd.name() {
+                        info!("Successfully added '{}'", name)
+                    };
+                }
+            }
+            other => {
+                return Err(anyhow!(
+                    "testsys has not yet added support for the '{}' variant family",
+                    other
+                ))
+            }
+        };
+
+        Ok(())
+    }
+}
+
+fn parse_key_val(s: &str) -> Result<(String, SecretName)> {
+    let mut iter = s.splitn(2, '=');
+    let key = iter.next().context("Key is missing")?;
+    let value = iter.next().context("Value is missing")?;
+    Ok((key.to_string(), SecretName::new(value)?))
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub(crate) enum TestType {
+    /// Run conformance testing on a given arch and variant
```

Comment:
I think these could be more generic and more specific at the same time:

```suggestion
    /// Conformance testing is a full integration test that asserts that Bottlerocket is working for
    /// customer workloads. For k8s variants, for example, this will run the full suite of sonobuoy
    /// conformance tests.
```